### ISSUE TITLE
Email notification conditions when using checkboxes

### ISSUE THREAD

**Issue by gglnx** [verbb/formie#477]
**Description**

I don't know if this a bug or works as intended: If I setup an email notification based on a condition, that a checkbox option is selected (`is`), the email notification will be not triggered, if other options in this checkbox field are selected too. It works if I'm using the condition `contains` and manually entering the checkbox option value.

It's possible the problem also occurs with other fields, that allow multiple values.

**Steps to reproduce**
1. Create a checkbox field with two options: `Option A`,`Option B`.
2. Create an email notification with the condition: Field = `Checkbox field`, Condition = `is`, Value = `Option A`.
3. Submit the form from the frontend with Option A selected, the email notification will be send.
4. Submit the form from the frontend with both options selected, the email notification will be **not** send.

### Form settings

- Multi-page form: No
- Submission Method: Page Reload
- Client-side Validation: No
- Custom Form Templates: No

**Additional info**
- Plugin version: 1.4.4
- Craft version: Craft 3.6.15
- Multi-site: No

**Comment by engram-design** [comment]
Yeah, so this is an interesting one, and I'm _slightly_ leaning towards this as being expected behaviour.

The reason for this is the `is` condition is an exact equal condition. Does a field value of `Option A and Option B` equal `Option A` - that would be no, because two values (whatever they might be) don't equate to the single value.

Whereas `contains` (as you've found) works due to it saying is `Option A` in this field value of `Option A and Option B`.

So it's working as expected at the moment. But it does raise another point as an aside, if we wanted to check if the field value was `Option A` and `Option B`, we'd have to create two new rows in the conditions table and set the `All` rules match condition. But that might not work with any other condition rules in the table.

So for the moment, continue using `contains` for checkboxes, unless you want exact matching (where the user picks _only_ that option). We might need to introduce being able to select multiple values for fields that allow multiple values.

**Comment by gglnx** [comment]
I would agree too that the `is` condition checks for an equal match. `Option A` is not the same as `[Option A, Option B]`. Maybe this should be documented in the interface itself. To support more use cases like `this two checkbox options needed be selected both` indeed switch for fields with multiple values to a `select[multiple]`.

My only problem with using `contains` as an alternative is, that this requries to reenter the values manually. I think the best approach would be to add a new condition called `has` that uses a select with all options and checks if `Option A` is in `[Option A, Option B]`.

**Comment by gglnx** [comment]
(It's possible `has` as a condition name doesn't make any sense and I'm just applying German sentence building to it.)

**Comment by engram-design** [comment]
I don't mind the `contains` one being free text, as it provides a bit more flexibility. I might have a bit more of a thought here to tackle two things at once. I might leave the `is` how it is now, but provide multiple options to check the value against. What if we do want to check if `Option A` and `Option B` are checked, in a field with say, 6 other options.

I'll likely introduce some UI to select multiple options, for the fields that makes sense.

**Comment by engram-design** [comment]
After further discussion and thoughts in https://github.com/verbb/formie/issues/620 I'm fixing this issue earlier than the 1.5.x milestone. I believe it makes sense to treat `is/is not` for checkboxes and other multi-value enabled fields as contains under the hood. 

Where `contains` will still work for free-text containing text. Say you have:

- Department A
- Department B
- Office A
- Office B

It might be preferable to setup a condition where the checkboxes field value `contains Department` as a string, rather than saying `= Department A + = Department B, etc`.

To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.4.22",
  "...": "..."
}
```

Then run `composer update`.

**Comment by engram-design** [comment]
Fixed in [1.4.23](https://github.com/verbb/formie/releases/tag/1.4.23)
